### PR TITLE
CIV-13865 Response application navigation

### PR DESCRIPTION
**Pull request by portizdev**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-13865

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by neeta-hmcts — DISMISSED** [review]

**Review by paul-pearson — DISMISSED** [review]

File: src/main/services/features/generalApplication/response/generalApplicationResponseService.ts [hunk 1]
```diff
@@ -31,12 +32,12 @@ export const saveRespondentAgreeToOrder = async (redisKey: string, agreeToOrder:
   }
 };
 
-export function getRespondToApplicationCaption(claim: Claim, gaApplicationId: string, lng: string): string {
-  const selectedApplication = claim.respondentGaAppDetails.find(application => application.gaApplicationId === gaApplicationId);
-  if (selectedApplication.generalAppTypes?.length > 1) {
+export function getRespondToApplicationCaption(generalAppTypes: ApplicationTypeOption[], lng: string): string {
+  //const selectedApplication = claim.respondentGaAppDetails.find(application => application.gaApplicationId === gaApplicationId);
```
Comment (manish14836) [inline src/main/services/features/generalApplication/response/generalApplicationResponseService.ts:36]: remove this comment

File: src/main/routes/features/generalApplication/response/viewApplicationController.ts [hunk 2]
```diff
@@ -1,37 +1,65 @@
...
 }) as RequestHandler);
 
+async function getRedirectUrl(applicationResponse: ApplicationResponse, applicationId: string, claimId: string) {
+  const claimantRespondingToDefendantVaryAJudgment = applicationResponse.case_data.generalAppType.types.includes(ApplicationTypeOption.VARY_PAYMENT_TERMS_OF_JUDGMENT);
+  const withConsent = applicationResponse.case_data.generalAppRespondentAgreement.hasAgreed === YesNoUpperCamelCase.YES;
+  if(claimantRespondingToDefendantVaryAJudgment){
+    return constructResponseUrlWithIdAndAppIdParams(claimId, applicationId, GA_ACCEPT_DEFENDANT_OFFER_URL);
+  }else if(withConsent){
+    return constructResponseUrlWithIdAndAppIdParams(claimId, applicationId, GA_AGREE_TO_ORDER_URL);
+  }else{
```
Comment (manish14836) [inline src/main/routes/features/generalApplication/response/viewApplicationController.ts:61]: No need of else

File: src/main/routes/features/generalApplication/response/respondentHearingPreferenceController.ts [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
 respondentHearingPreferenceController.get(GA_RESPONDENT_HEARING_PREFERENCE_URL, (async (req: AppRequest, res: Response, next: NextFunction) => {
   try {
     const claimId = req.params.id;
-    const claim = await getClaimById(claimId, req, true);
+    const gaResponse = await getDraftGARespondentResponse(generateRedisKeyForGA(<AppRequest>req));
     const lang = req.query.lang ? req.query.lang : req.cookies.lang;
-    const applicationType: string = getRespondToApplicationCaption(claim, req.params.appId, lang);
-    const continueLinkUrl = constructResponseUrlWithIdAndAppIdParams(claimId, req.params.appId, GA_RESPONDENT_WANT_TO_UPLOAD_DOCUMENT_URL); // TODO: add url
+    const applicationType: string = getRespondToApplicationCaption(gaResponse.generalApplicationType, lang);
+    const continueLinkUrl = constructResponseUrlWithIdAndAppIdParams(claimId, req.params.appId, GA_RESPONSE_HEARING_ARRANGEMENT_URL  );
```
Comment (manish14836) [inline src/main/routes/features/generalApplication/response/respondentHearingPreferenceController.ts:25]: Check formatting space before bracket. Not sure if it is in github only.

File: src/main/routes/features/generalApplication/response/respondentAgreementController.ts [hunk 4]
```diff
@@ -44,16 +51,17 @@ respondentAgreementController.post(GA_RESPONDENT_AGREEMENT_URL, (async (req: App
...
       const redisKey = generateRedisKey(req);
       const claim = await getCaseDataFromStore(redisKey);
+      const gaResponse = await getDraftGARespondentResponse(generateRedisKeyForGA(req));
       const lang = req.query.lang || req.cookies.lang;
-      return await renderView(claimId, claim, form, lang, req.params.appId, res);
+      return await renderView(claimId, claim, form, lang, req.params.appId, gaResponse, res);
     }
     await saveRespondentAgreement(generateRedisKeyForGA(req), respondentAgreement);
-    res.redirect('test'); // TODO: add url
+    res.redirect(constructResponseUrlWithIdAndAppIdParams(claimId, req.params.appId, GA_RESPONDENT_WANT_TO_UPLOAD_DOCUMENT_URL)); // TODO: add url
```
Comment (manish14836) [inline src/main/routes/features/generalApplication/response/respondentAgreementController.ts:64]: Remove // TODO: add url. It is present in multiple places on newly added lines.

File: src/main/routes/features/generalApplication/response/respondentAgreementController.ts [hunk 5]
```diff
@@ -2,24 +2,31 @@ import {NextFunction, RequestHandler, Response, Router} from 'express';
...
+import {constructResponseUrlWithIdAndAppIdParams} from 'common/utils/urlFormatter';
 
 const respondentAgreementController = Router();
 const viewPath = 'features/generalApplication/respondent-agreement';
 
-const renderView = async (claimId: string, claim: Claim, form: GenericForm<RespondentAgreement>, lng: string, appId: string, res: Response): Promise<void> => {
+const renderView = async (claimId: string, claim: Claim, form: GenericForm<RespondentAgreement>, lng: string, appId: string, gaResponse: GaResponse, res: Response): Promise<void> => {
   const cancelUrl = await getCancelUrl(claimId, claim);
-  const backLinkUrl = 'test'; // TODO: add url
+  const backLinkUrl = !gaResponse.agreeToOrder ? constructResponseUrlWithIdAndAppIdParams(claimId, appId, GA_RESPONSE_VIEW_APPLICATION_URL) : constructResponseUrlWithIdAndAppIdParams(claimId, appId, GA_AGREE_TO_ORDER_URL); // TODO: add url
```
Comment (manish14836) [inline src/main/routes/features/generalApplication/response/respondentAgreementController.ts:26]: Remove // TODO: add url. It is present in multiple places on newly added lines.

File: src/main/routes/features/generalApplication/response/checkAnswersResponseController.ts [hunk 6]
```diff
@@ -60,17 +62,17 @@ gaCheckAnswersResponseController.post(GA_RESPONSE_CHECK_ANSWERS_URL, (async (req
     } else {
       const redisKey = generateRedisKeyForGA(req);
       await saveRespondentStatementOfTruth(redisKey, statementOfTruth);
-      res.redirect('test'); // TODO: correct URL      
+      res.redirect('test'); // TODO: correct URL
```
Comment (manish14836) [inline src/main/routes/features/generalApplication/response/checkAnswersResponseController.ts:65]: Is this not in scope of this ticket?

File: src/main/routes/features/generalApplication/respondentAgreeToOrderController.ts [hunk 7]
```diff
@@ -41,23 +51,31 @@ respondentAgreeToOrderController.post(GA_AGREE_TO_ORDER_URL, (async (req: AppReq
...
-      res.redirect('test_url');
+      res.redirect(getRedirectUrl(claimId, applicationId, req.body.option));
     }
   } catch (error) {
     next(error);
   }
 }) as RequestHandler);
 
+function getRedirectUrl(claimId: string, applicationId: string, option: YesNo.YES) {
+  return option == YesNo.YES ? constructResponseUrlWithIdAndAppIdParams(claimId, applicationId,  GA_RESPONDENT_WANT_TO_UPLOAD_DOCUMENT_URL) : constructResponseUrlWithIdAndAppIdParams(claimId, applicationId, GA_RESPONDENT_AGREEMENT_URL);
```
Comment (manish14836) [inline src/main/routes/features/generalApplication/respondentAgreeToOrderController.ts:78]: Use ===

File: src/main/services/features/generalApplication/response/generalApplicationResponseService.ts (hunk 1 above)
Comment (jeswanth-hmcts) [reply src/main/services/features/generalApplication/response/generalApplicationResponseService.ts]: removed the comment

File: src/main/routes/features/generalApplication/response/respondentAgreementController.ts (hunk 4 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/generalApplication/response/respondentAgreementController.ts]: removed the todo

File: src/main/routes/features/generalApplication/response/respondentAgreementController.ts (hunk 5 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/generalApplication/response/respondentAgreementController.ts]: removed todo

File: src/main/routes/features/generalApplication/response/checkAnswersResponseController.ts (hunk 6 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/generalApplication/response/checkAnswersResponseController.ts:65]: no

File: src/main/routes/features/generalApplication/respondentAgreeToOrderController.ts (hunk 7 above)
Comment (jeswanth-hmcts) [reply src/main/routes/features/generalApplication/respondentAgreeToOrderController.ts]: used ===

**Review by manish14836 — APPROVED** [review]

**Review by kenneth-hmcts — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]
